crates/red_knot_python_semantic/src/program.rs

```diff
@@ -54,21 +54,23 @@ impl Program {
 }
 
 #[derive(Clone, Debug, Eq, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize))]
 pub struct ProgramSettings {
     pub target_version: PythonVersion,
     pub search_paths: SearchPathSettings,
 }
 
 /// Configures the search paths for module resolution.
 #[derive(Eq, PartialEq, Debug, Clone)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize))]
 pub struct SearchPathSettings {
     /// List of user-provided paths that should take first priority in the module resolution.
     /// Examples in other type checkers are mypy's MYPYPATH environment variable,
     /// or pyright's stubPath configuration setting.
     pub extra_paths: Vec<SystemPathBuf>,
 
     /// The root of the workspace, used for finding first-party modules.
-    pub src_root: SystemPathBuf,
+    pub workspace_root: SystemPathBuf,
```

Comment:
I did ask about this in Discord because I felt we discussed this before and it was unclear to me how it should work. @AlexWaygood then gave me a reply that seemed reasonable to me, see https://discord.com/channels/1039017663004942429/1228460843033821285/1304410214757433364

Generally, this PR doesn't change how we resolve the search path settings. It keeps doing what we used to do and I also don't mind reverting the rename of the `SearchPathSettings::workspace_root`. 

> A related question is how workspace members' code is expected to get onto the import search path. Do we offer some red-knot-specific functionality to make this happen automatically, or do we require that either workspace members are editable-installed into the venv (and thus we find them via .pth files) or else you have to use extra_paths explicitly for them?

I understand that we rely on the package manager to put projects on the path.

Reply:
I agree with everything @AlexWaygood says in the first two paragraphs of that Discord comment. I'm not sure about this part:

> I think we should probably rename the first-party search path; calling it src is kind of confusing. It's not meant to correspond to the src directory inside a workspace; it's meant to correspond to the workspace root

In a project layout where the Python packages all go inside a `src/` subdirectory, at runtime the project must be installed to be importable, and usually it is run via some installed command/script, and `.` is not actually added to the import search path in that case. Which is a good thing, because if it were, you'd end up with overlapping search paths and all your modules importable as both `foo` and `src.foo`, which is not good, especially at runtime.

It is not clear to me how we can detect this project structure and avoid putting the workspace root on the search path in this case. I guess one option is just to punt on it and hope nobody is affected by the fact that `src.foo` imports are wrongly allowed.

> I understand that we rely on the package manager to put projects on the path.

👍

Reply:
I guess I'm also not clear that it's right to have only one "first-party" search path. I guess it partly depends what we mean by "first-party", but to the extent that it influences things like "which code do we show diagnostics for", I think it is important that all code within the workspace root (even if it is discovered via editable-install `.pth` files in the venv) should be considered as "first-party".

Reply:
> I guess I'm also not clear that it's right to have only one "first-party" search path. I guess it partly depends what we mean by "first-party", but to the extent that it influences things like "which code do we show diagnostics for", I think it is important that all code within the workspace root (even if it is discovered via editable-install `.pth` files in the venv) should be considered as "first-party".

I suppose that depends what we're using the "first-party search path" for. If we're using it as the canonical definition for what constitutes "first-party" code, then what you say makes sense. But _do_ we need to tie those two concepts together? (I guess it would simplify some things?)

Reply:
Agreed. If we're going to go with the "there should only ever be one of this kind of search path", I'd rename it everywhere to `workspace_root` or just `root`, rather than `first_party`.